### DISCUSSION TITLE
Update from Firmware: 2022-12-20

### DISCUSSION THREAD

**Discussion by msggsm** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#527] [Q&A]
Hello,
I have a question about updating my device:
I bought a used Hawaii Hydrojet Pro (new gray square pump) with the ESP8266 PCB_V2B  already installed and working.
In the menu i can see that it is Firmware: 2022-12-20 with Model set to: MALDIVES2021,
In the menu I can find the File Uploader, and I found here in FAQs that I would have to upload new firmware AND files.
As I did not set up the device I don't have the files.
Can I somehow backup/download them from the running device or how could I upgrade to the latest version?
Thanks in advance,
Mark

**Comment by visualapproach** [comment]
Hi, I suggest you read the manual first to get a grip on the procedure. It will be easier to update from v4 (2023...) but it's not rocket science from v3 either.

**Comment by SRBT1408** [comment]
I have read the user manual. Now I have a few questions about the update, I'm on version 2022-12-20 with model set to: MALDIVES202. Can I just upload files or do I have to flash the new firmware to get OTA?

**Reply by visualapproach** [reply]
Updating is not a feature per se. You do it as you did it the first time (as described in the manual).  Download project, open in platformio and build+upload. Also build+upload the filesystem. It will erase previous install. If you want to save your old saved files (settings etc) you need to not build filesystem but upload every data file one at the time from the web ui.
From v4 there is a web ui button that does all that for you.
Good luck
